src/refspec.c

```diff
@@ -160,6 +160,35 @@ void git_refspec__free(git_refspec *refspec)
 	memset(refspec, 0x0, sizeof(git_refspec));
 }
 
+int git_refspec_parse(git_refspec **refspec, const char *input, int is_fetch) {
+	git_refspec *new_refspec;
+	if (refspec == NULL || input == NULL) {
+		return -1;
+	}
+
+	is_fetch = !!is_fetch;
+	new_refspec = git__malloc(sizeof(git_refspec));
+	if (new_refspec == NULL) {
+		return -1;
+	}
+
+	if (!git_refspec__parse(new_refspec, input, is_fetch)) {
+		git__free(new_refspec);
+		return -1;
+	}
+
+	*refspec = new_refspec;
+	return 0;
+}
+
+void git_refspec_free(git_refspec *refspec) {
+	if (refspec == NULL) {
```

Comment:
I'd be tempted to say that this test is not needed : `git_refspec__free` already handles it, and `git__free` doesn't care about `NULL`.

Reply:
done